src/main/java/duke/task/TaskListParser.java

```diff
@@ -0,0 +1,85 @@
+package duke.task;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * Represents a parser that parses a task list from a file.
+ */
+public class TaskListParser {
+    /**
+     * Parses a task list from a file.
+     *
+     * @param file The file to parse from.
+     * @return The parsed task list.
+     * @throws IOException                If an I/O error occurs.
+     * @throws InvalidDataFormatException If the data file is not in the correct
+     *                                    format.
+     * @throws DukeDateTimeParseException If the date and time is not in the correct
+     *                                    format.
+     */
+    public static TaskList parse(File file) throws IOException, InvalidDataFormatException, DukeDateTimeParseException {
```

Comment:
Perhaps method name could be more informative and detailed like parseListFromFile or parseFile? Similar to how you did it for parseTask.

Reply:
Good point, I will do that!